src/Avalonia.Base/Data/Core/Plugins/ObservableStreamPlugin.cs

```diff
@@ -54,56 +61,32 @@ public virtual bool Match(WeakReference<object?> reference)
                   x.IsGenericType &&
                   x.GetGenericTypeDefinition() == typeof(IObservable<>)).GetGenericArguments()[0];
 
-            // Get the Observable.Select method.
-            var select = GetObservableSelect(sourceType);
-
-            // Make a Box<> delegate of the correct type.
-            var funcType = typeof(Func<,>).MakeGenericType(sourceType, typeof(object));
-            var box = GetType().GetMethod(nameof(Box), BindingFlags.Static | BindingFlags.NonPublic)!
-                .MakeGenericMethod(sourceType)
-                .CreateDelegate(funcType);
+            // Get the BoxObservable<T> method.
+            var select = GetBoxObservable(sourceType);
 
-            // Call Observable.Select(target, box);
+            // Call BoxObservable(target);
             return (IObservable<object?>)select.Invoke(
                 null,
-                new object[] { target, box })!;
+                new[] { target })!;
         }
 
         [RequiresUnreferencedCode(TrimmingMessages.StreamPluginRequiresUnreferencedCodeMessage)]
-        private static MethodInfo GetObservableSelect(Type source)
+        private static MethodInfo GetBoxObservable(Type source)
         {
-            return GetObservableSelect().MakeGenericMethod(source, typeof(object));
+            return (s_observableGeneric ??= GetBoxObservable()).MakeGenericMethod(source);
         }
 
-        private static MethodInfo GetObservableSelect()
+        [RequiresUnreferencedCode(TrimmingMessages.StreamPluginRequiresUnreferencedCodeMessage)]
+        private static MethodInfo GetBoxObservable()
         {
-            if (observableSelect == null)
-            {
-                observableSelect = typeof(Observable).GetRuntimeMethods().First(x =>
-                {
-                    if (x.Name == nameof(Observable.Select) &&
-                        x.ContainsGenericParameters &&
-                        x.GetGenericArguments().Length == 2)
-                    {
-                        var parameters = x.GetParameters();
-
-                        if (parameters.Length == 2 &&
-                            parameters[0].ParameterType.IsConstructedGenericType &&
-                            parameters[0].ParameterType.GetGenericTypeDefinition() == typeof(IObservable<>) &&
-                            parameters[1].ParameterType.IsConstructedGenericType &&
-                            parameters[1].ParameterType.GetGenericTypeDefinition() == typeof(Func<,>))
-                        {
-                            return true;
-                        }
-                    }
-
-                    return false;
-                });
-            }
-
-            return observableSelect;
+            return s_observableSelect
+               ??= typeof(ObservableStreamPlugin).GetMethod(nameof(BoxObservable), BindingFlags.Static | BindingFlags.NonPublic)
```

Comment:
Previously this plugin was searching for Select method from System.Reactive package. Now it uses more specialized method defined in the same class and pinned with DynamicDependency, so it's AOT safe.